public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -279,6 +283,15 @@ export function PrometheusDatasource(instanceSettings, $q, backendSrv, templateS
     return metricName + '{' + labelPart + '}';
   };
 
+  this.alignTime = function(time, range) {
```

Comment:
I understand what is the problem.
Change implementation.
If refresh period is shorter than 60sec, use refresh period as align interval.